provisioner/provision.js

```diff
@@ -198,6 +203,37 @@ Provisioner.prototype.runAllProvisionersOnce = function () {
 };
 
 /**
+ * Figure out how to create the launch information based on a bid then
+ * insert the secrets into the secret storage
+ */
+Provisioner.prototype.spawn = function (workerType, bid) {
+  assert(workerType);
+  assert(bid);
+
+  var launchInfo = workerType.createLaunchSpec(bid.region, bid.type);
+
+  // This should probably move from here to createLaunchSpec but that
+  // can happen later
+  launchInfo.launchSpec.Placement = {
+    AvailabilityZone: bid.zone,
+  };
+
+  var p = this.Secret.create({
+    token: launchInfo.securityToken,
+    workerType: workerType.workerType,
+    secrets: launchInfo.secrets,
+    scopes: launchInfo.scopes,
+    expiration: taskcluster.fromNow('40 mins'),
+  });
+
+  p = p.then(()=> {
+    this.awsManager.requestSpotInstance(launchInfo, bid);
+  });
+
+  return p;
```

Comment:
How about:
```js
let secret = await this.Secret.create({
  token: launchInfo.securityToken,
  workerType: workerType.workerType,
  secrets: launchInfo.secrets,
  scopes: launchInfo.scopes,
  expiration: taskcluster.fromNow('40 mins'),
  spotRequestId: '' // Empty string to indicate we don't have any
});

var spotReqId = await this.awsManager.requestSpotInstance(launchInfo, bid);

await secret.modify(secret => {
  secret.spotRequestId = spotReqId;
});
```

Now when we get a callback for the secret, if `secret.spotRequestId !== ''` we look for the spotRequest and submit duration stats... If `secret.spotRequestId == ''` then we don't try to correlate... This will happen very very rarely.